pkgs/applications/science/molecular-dynamics/lammps/default.nix

```
@@ -1,54 +1,76 @@
-{ lib, stdenv, fetchFromGitHub
-, libpng, gzip, fftw, blas, lapack
+{ lib
+, stdenv
+, fetchFromGitHub
+, libpng
+, gzip
+, fftw
+, blas
+, lapack
 , withMPI ? false
 , mpi
+, cmake
+# Available list of packages can be found near here:
+# https://github.com/lammps/lammps/blob/develop/cmake/CMakeLists.txt#L222
+, packages ? {
+  ASPHERE = true;
+  BODY = true;
+  CLASS2 = true;
+  COLLOID = true;
+  COMPRESS = true;
+  CORESHELL = true;
+  DIPOLE = true;
+  GRANULAR = true;
+  KSPACE = true;
+  MANYBODY = true;
+  MC = true;
+  MISC = true;
+  MOLECULE = true;
+  OPT = true;
+  PERI = true;
+  QEQ = true;
+  REPLICA = true;
+  RIGID = true;
+  SHOCK = true;
+  ML-SNAP = true;
+  SRD = true;
+  REAXFF = true;
+}
 }:
```

To make sure we have right API as incompatibilities can not be detected at compile time:

```suggestion
}:

assert blas.isILP64 == lapack.isILP64;
assert ! blas.isILP64;
```

---

> To make sure we have right API as incompatibilities can not be detected at compile time:

This wasn't there before, but I'll add it. Allow me to put that in `meta.broken` instead of straight at the top, since I've heard often complaints that this may block evaluation to aggressively.

---

> > This wasn't there before, but I'll add it.
> 
> Just to give some context here: it is not consistently implement everywhere as the problem with LP64/ILP64 in Fortran is often overlooked, but I hope to get this step-by-step in more places.
> 
> > Allow me to put that in meta.broken instead of straight at the top, since I've heard often complaints that this may block evaluation too aggressively.
> 
> OK, that is a valid solution. In standard nixpkgs this should not be a problem though, as one needs to override this explicitly (see [nixos.org/manual/nixpkgs/stable/#sec-overlays-alternatives-blas-lapack](https://nixos.org/manual/nixpkgs/stable/#sec-overlays-alternatives-blas-lapack)).

I agree. I found the link to the topic discussing this: https://github.com/NixOS/nixpkgs/issues/36229 . Although this issue is closed, [people still refer to it](https://github.com/NixOS/nixpkgs/pull/230131#discussion_r1189493257).